packages/cf-component-table/src/Table.js

```diff
@@ -1,41 +1,90 @@
 import React from 'react';
-
 import PropTypes from 'prop-types';
+import {
+  mapChildren,
+  filterNone,
+  applyTheme,
+  createComponent,
+  connectStyles
+} from 'cf-style-container';
+import TableTheme from './TableTheme.js';
+
+const TableStyles = ({ theme, bordered }) =>
+  filterNone({
+    background: theme.colorWhite,
+    borderCollapse: bordered ? 'separate' : 'collapse',
+    borderSpacing: 0,
+    maxWidth: '100%',
+    width: '100%',
+
+    // Table styles
+    border: bordered ? '1px solid ${theme.tableBorderColor}' : undefined,
+    borderLeft: bordered ? 0 : undefined,
+
+    [`@media (max-width: ${theme.breakpoints.tablet})`]: {
+      display: 'block',
+      width: '100%',
+      boxSizing: 'border-box'
+    },
+
+    [`@media screen and (max-width: ${theme.breakpoints.tablet})`]: {
+      display: 'block'
+    }
+  });
 
 class Table extends React.Component {
   render() {
-    let className = 'cf-table';
-
-    if (this.props.striped) className += ' cf-table--striped';
-    if (this.props.hover) className += ' cf-table--hover';
-    if (this.props.bordered) className += ' cf-table--bordered';
-    if (this.props.condensed) className += ' cf-table--condensed';
-    if (this.props.className.trim())
-      className += ' ' + this.props.className.trim();
-
-    return (
-      <table className={className}>
-        {this.props.children}
-      </table>
+    const {
+      is,
+      className,
+      striped,
+      hover,
+      bordered,
+      condensed,
+      bare,
+      children
+    } = this.props;
+
+    return React.createElement(
+      is,
+      { className },
+      mapChildren(children, child => {
+        if (React.isValidElement(child)) {
+          return React.cloneElement(child, {
+            condensed,
+            striped,
+            bordered,
+            hover,
+            bare
+          });
+        }
+        return child;
+      })
     );
   }
 }
 
 Table.propTypes = {
-  className: PropTypes.string,
+  is: PropTypes.oneOf(['table', 'div']),
+  className: PropTypes.string.isRequired,
   striped: PropTypes.bool,
   hover: PropTypes.bool,
   bordered: PropTypes.bool,
   condensed: PropTypes.bool,
+  bare: PropTypes.bool,
   children: PropTypes.node
 };
 
 Table.defaultProps = {
-  className: '',
+  is: 'table',
   striped: false,
   hover: false,
   bordered: true,
-  condensed: false
+  condensed: false,
+  bare: false
 };
 
-export default Table;
+export const createTable = (...styles) =>
```

Comment:
I haven't had the time to go thru this in detail, so I'll just briefly summary what's in my head now.

First, overriding styles externally is an inevitability, so it's a necessity, I don't think we disagree on this.

The disagreement mainly is the degree we allow customization. You seem to be misled that whitelisting style props in a theme makes it harder to override a style prop that's not allowed in a theme. Try this:

```jsx
import { Button } from 'cf-component-button';
const MyButton = createComponent(() => ({background: 'black'}), Button);
``

This button is themed, and `background` [is not whitelisted](https://github.com/cloudflare/cf-ui/blob/master/packages/cf-component-button/src/Button.js). What actually gets generated is 2 fela generate CSS rule names applied to the button like this:

```html
<style>
.cf-14a9t3k {
  /* the original backgroundColor prop got DCE'ed and disappeared */
}
.cf-1e4pz7w {
  background: 'color';
}
</style>
<button type="button" class="cf-14a9t3k cf-1e4pz7w">
```

Even if fela didn't do this, overriding the button's style is not hard at all. I can do a number of things similar to this:

```jsx
class MyButton extends React.Component {
    componentWillMount() {
      const style = document.createElement('style');
      style.innerText = ".my-button { background-color; 'black' !important }";
      document.body.insertBefore(style, document.body.firstChild);
    }
    render() {
        return (
          <div className="my-button">
            <Button>
          </div>
        );
    }
}
```

There are 2 points I want to get across.

1. There's no way you can prevent me from overriding any style I want, when I want, and you are not making it any harder either.
2. Themes, are used as place holders for values derived from the base theme, and nothing more. We don't ever override these theme values in our existing SASS and PostCSS stylesheets.

There's more I have to say about themes in our ported components now, like the button. We are essentially abusing it as a style function. I'll save this for another day as this reply is already too long.